server/database.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * @fileoverview database function handler
+ */
+var assert = require('assert');
+
+var log = require('./log.js');
+
+/** @type {database} */
+module.exports = database;
+
+/**
+ * @constructor
+ */
+function database() {}
+
+/**
+ * Checks the user information for signing in.
+ * @param {!Object} collection Database collection.
+ * @param {!Object} query Database query.
```

Comment:
As far as I remember it could be possible that mongodb queries are recursively defined. You may want to reflect that as well.